src/Angor/Avalonia/Angor.Contexts.Wallet/Domain/BroadcastedTransactionExtensions.cs

```diff
@@ -0,0 +1,37 @@
+namespace Angor.Contexts.Wallet.Domain;
+
+public static class BroadcastedTransactionExtensions
+{
+    public static Balance GetBalance(this BroadcastedTransaction tx)
+    {
+        long totalInputs  = tx.WalletInputs .Sum(i => i.Amount.Sats);
+        long totalOutputs = tx.WalletOutputs.Sum(o => o.Amount.Sats);
+        return new Balance(totalOutputs - totalInputs);
+    }
+    
+    public static Amount GetTotalSent(this BroadcastedTransaction tx)
+        => new Amount(tx.AllOutputs
+            .Where(o => tx.WalletOutputs.All(w => w.Address != o.Address))
+            .Sum(o => o.Amount.Sats));
+    
+    public static Amount GetTotalReceived(this BroadcastedTransaction tx)
+        => new Amount(tx.WalletOutputs.Sum(o => o.Amount.Sats));
+    
+    public static IReadOnlyCollection<Address> GetSenderAddresses(
```

Comment:
Could use some description here, are you only taking inputs that do not belong to the wallet?

Reply:
Yes, I'll add XML documentation to clarify these methods.

The `GetTotalSent` method calculates the total amount sent to external addresses. It uses `tx.AllOutputs` and filters them to only include outputs where the address is not in any of the wallet's own outputs (`.Where(o => tx.WalletOutputs.All(w => w.Address != o.Address))`).

This distinction is important because:
- `WalletInputs` and `WalletOutputs` represent the wallet's own transactions
- `AllInputs` and `AllOutputs` include all transaction parts, both the wallet's and external ones

I'll add proper documentation to make these calculations clearer.